bundle/src/main/java/com/adobe/acs/commons/mcp/impl/processes/BulkPageTagger.java

```diff
@@ -0,0 +1,254 @@
+    private static final long serialVersionUID = 798823856839772874L;
+
+    /**
+     * The constant NAME.
+     */
+    public static final String NAME = "Bulk Page Tagger";
+
+    /**
+     * The Excel file.
+     */
+    @FormField(
+            name = "Excel File",
+            description = "Provide the .xlsx file that defines the content pages and the corresponding cq:tags to be added on the pages",
+            component = FileUploadComponent.class,
+            options = {"mimeTypes=application/vnd.openxmlformats-officedocument.spreadsheetml.sheet", "required"}
+    )
+    public transient InputStream excelFile = null;
+
+
+    @Override
+    public void init() throws RepositoryException {
+        // Nothing to be done here.
+
+    }
+
+
+    @Override
+    public void buildProcess(ProcessInstance instance, ResourceResolver rr) throws LoginException {
+        report.setName(instance.getName());
+        instance.getInfo().setDescription("Bulk Tag AEM content Pages");
+        instance.defineCriticalAction("Parse Excel File", rr, this::parseExcel);
+        instance.defineCriticalAction("Add Tags to Content Pages", rr, this::tagPages);
+
+    }
+
+
+    /**
+     * The Page tag mapping dataStructure.
+     */
+    transient volatile HashMap<String, String> pageTagMapping = new LinkedHashMap<>();
+
+    /**
+     * Parse input excel.
+     *
+     * @param manager the manager
+     * @throws Exception the exception
+     */
+    @SuppressWarnings({"squid:S3776", "squid:S1141"})
+    public void parseExcel(ActionManager manager) throws Exception {
+        manager.withResolver(rr -> {
+            final XSSFWorkbook workbook = new XSSFWorkbook(excelFile);
+            final XSSFSheet sheet = workbook.getSheetAt(0);
+
+            final Iterator<Row> rows = sheet.rowIterator();
+            final String tagsRootPath = new TagCreator.TagRootResolver(rr).getTagsLocationPath();
+
+
+            if (tagsRootPath == null) {
+                recordAction(ReportRowSatus.FAILED_TO_PARSE,
+                        "Abandoning Tag parsing. Unable to determine AEM Tags root (/content/cq:tags vs /etc/tags). Please ensure the path exists and is accessible by the user running Tag Creator.", "N/A");
+                return;
+            }
+
+            while (rows.hasNext()) {
+                final Row row = rows.next();
+                if (row.getCell(0) == null) {
+                    break;
+                }
+                if (row.getRowNum() != 0 && row.getCell(0) != null) {
+                    pageTagMapping.put(row.getCell(0).getStringCellValue(), row.getCell(1).getStringCellValue());
+                }
+            }
+
+        });
+
+    }
+
+
+    /**
+     * Tag pages from the excel file with cq:tags.
+     *
+     * @param manager the manager
+     * @throws Exception the exception
+     */
+    @SuppressWarnings({"squid:S3776", "squid:S1141"})
+    public void tagPages(ActionManager manager) throws Exception {
```

Comment:
please add  @SuppressWarnings("squid:S112") to suppress the "Define and throw a dedicated exception instead of using a generic one" Sonar warning.